api/src/main/java/jakarta/persistence/EntityManagerFactory.java

```diff
@@ -371,6 +372,36 @@ public interface EntityManagerFactory extends AutoCloseable {
      */
     <E> Map<String, EntityGraph<? extends E>> getNamedEntityGraphs(Class<E> entityType);
 
+    /**
+     * Register a listener for the given kind of entity lifecycle event
+     * and given entity type, returning an {@link EntityListenerRegistration}
+     * which may be used to {@linkplain EntityListenerRegistration#cancel cancel}
+     * the registration.
+     * <p>
+     * The given entity type may be an entity class or any supertype of
+     * an entity class, including {@code java.lang.Object}. The lifecycle
+     * event type is identified by the class of a callback annotation type,
+     * for example, {@link PostPersist PostPersist.class}. When a lifecycle
+     * event of the given type occurs for any entity assignable to the given
+     * entity type, the listener is invoked with the entity instance.
```

Comment:
The user will want to know if the listener might be invoked under the same transaction a the entity lifecycle operation, such that transactional operations performed by a listener would be rolled back/committed along with the lifecycle operation versus whether they happen independently of it.
It should be stated whether or not the listener is allowed to modify the supplied entity instance (and if so, what the consequence of it are), and whether or not it can be relied upon that the supplied entity instance reflects automatically generated or autoincremented values.

Reply:
Yeah, all this is already defined in the spec, but I agree it would be nice to have much more of that info in the javadoc. I actually already started work on that here: https://github.com/jakartaee/persistence/pull/642/files